great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -293,3 +297,58 @@ def root_directory(self) -> Optional[str]:
 
         """
         return self._context_root_directory
+
+    def _instantiate_datasource_from_config_and_update_project_config(
+        self,
+        name: str,
+        config: dict,
+        initialize: bool = True,
+        save_changes: bool = False,
+    ) -> Optional[Datasource]:
+        """Instantiate datasource and optionally persist datasource config to store and/or initialize datasource for use.
+
+        Args:
+            name: Desired name for the datasource.
+            config: Config for the datasource.
+            initialize: Whether to initialize the datasource or return None.
+            save_changes: Whether to save the datasource config to the configured Datasource store.
+
+        Returns:
+            If initialize=True return an instantiated Datasource object, else None.
+        """
+
+        datasource_config: DatasourceConfig = datasourceConfigSchema.load(config)
+
+        if save_changes:
+
+            datasource_config["name"] = name
+            resource_ref: GeCloudResourceRef = self._datasource_store.create(
+                datasource_config
+            )
+            datasource_config.id_ = resource_ref.ge_cloud_id
```

Comment:
This is where we add the id from the response on create from the cloud backend.

Reply:
love it!